docs/markdown/Wrap-dependency-system-manual.md

```diff
@@ -143,6 +143,34 @@ automatically by adding the following *(since 0.48.0)*:
 clone-recursive = true
 ```
 
+## Patch files
+
+*Since: 0.49.0*
+
+A list of local patch files can be provided in the `[patch-files]` section. They
+will be applied after the project has been extracted or cloned, and after the
+`patch_filename` has been applied. The `patch` command-line tool must be
+available.
+
+The `[patch-files]` section is optional and might contain the following keys:
+- `strip`: Number of leading components from file names to strip. Defaults to 1
```

Comment:
Shouldn't the strip level be able to be specified per patch?

Reply:
Hm, you're right and it's annoying because it makes the syntax more complicated :( Not sure it's worth the trouble, cerbero (from gst project) does not seems to be able to set strip for each patch, is it?

Reply:
Any suggestion on the format we should use in the wrap file to allow per patch strip value?

One section per patch seems a bit cumbersome but the most future proof?
```
[patch 0001-foo.patch]

[patch 0002-bar.patch]
strip = 1
```

Reply:
Or
```
[patch]
filename = 0001-foo.patch
[patch]
filename = 0002-bar.patch
strip = 1
```

Reply:
I would like to standardize on -p1 as `git format-patch` / `git diff` create by default, and forbid specifying this manually. What is the use case for allowing it to be configurable?

Reply:
Also, if we don't support configurable strip settings we don't need an entire section for this, we could just do:

```
[wrap-file]
diff_filename = ['0001-Change-return-value-to-43.patch',
                 '0002-Change-return-value-to-44.patch',
                 '0001-Change-foo-to-executable.patch',
                ]
```

Reply:
@eli-schwartz strip should definitely defaults to 1. The use-case for having other value is for projects from another century that are not in git in the first place, like svn or pure tarball, in which case you could have patches with strip=0. But I think it actually makes no sense to mix strip levels, my original PR is actually fine I think.

Reply:
Why would you have patches with strip=0? Just create them via `diff -ur old/ new/` :p purely tarball based patching is a complete nightmare otherwise.

Or `svn diff --git` which adds `a/` and `b/` at the start.

Reply:
I think a separate setting for `strip` is not necessary; as @eli-schwartz says, the user can convert a patch file, or regenerate it, if it's a non-standard legacy one.

`diff_filename` sounds like it mirrors `patch_filename` and `source_filename`. However, it accepts a list instead of a single filename. How about `diff_filename_list`? Or it could also accept an unquoted comma-separated list, which would make it compatible with `patch_filename` and `source_filename` (when providing a single element).